clients/ops/privacy-center/components/modals/PrivacyRequestForm.tsx

```diff
@@ -0,0 +1,297 @@
+import React, { useEffect, useState } from "react";
+import {
+  Button,
+  chakra,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Stack,
+  Text,
+} from "@fidesui/react";
+
+import { useFormik } from "formik";
+
+import type { AlertState } from "../../types/AlertState";
+
+import config from "../../config/config.json";
+
+import { ModalViews } from "./types";
+import { hostUrl } from "../../constants";
+
+const usePrivacyRequestForm = ({
+  onClose,
+  action,
+  setAlert,
+  setCurrentView,
+  setPrivacyRequestId,
+  isVerificationRequired,
+}: {
+  onClose: () => void;
+  action: typeof config.actions[0] | null;
+  setAlert: (state: AlertState) => void;
+  setCurrentView: (view: ModalViews) => void;
+  setPrivacyRequestId: (id: string) => void;
+  isVerificationRequired: boolean;
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const formik = useFormik({
+    initialValues: {
+      name: "",
+      email: "",
+      phone: "",
+    },
+    onSubmit: async (values) => {
+      if (!action) {
+        // somehow we've reached a broken state, return
+        return;
+      }
+
+      setIsLoading(true);
+
+      const body = [
+        {
+          identity: {
+            email: values.email,
+            phone_number: values.phone,
+            // enable this when name field is supported on the server
+            // name: values.name
+          },
+          policy_key: action.policy_key,
+        },
+      ];
+
+      const handleError = () => {
+        setAlert({
+          status: "error",
+          description: "Your request has failed. Please try again.",
+        });
+        onClose();
+      };
+
+      try {
+        const response = await fetch(`${hostUrl}/privacy-request`, {
+          method: "POST",
+          headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json",
+            "X-Fides-Source": "fidesops-privacy-center",
```

Comment:
We can. I just added it in. It would be nice to keep things consistent between the projects. eslint complained and said the [headers-polyfill](https://www.npmjs.com/package/headers-polyfill) package needed to be installed. They'll be one more small dependency added to the project. No big deal

Reply:
Thanks so much!